### PR TITLE
Refactoring base class

### PR DESCRIPTION
**Pull request by MKyhos**
Refactoring PR to correct some issues in the base classes (detected by @LadaRudnitckaia  and @hammti ), and to 
further improve testing coverage etc. To be merged with http testing (open pr #25 ) somehow.

### CODE REVIEW

**Comment by pr130** [comment]
made some small changes - otherwise looks good to me!
- removed unnecassary check for base_url (not required anymore because it is a required argument) 
- added a message that lets the user know that they won't be able to use certain functionalities if they don't specify base_url_v1. 
- split up tests referring to the construction of kobo into smaller test_thats. that is probably a matter of taste but from [R packages testing chapter](https://r-pkgs.org/tests.html#test-tests):

> It’s up to you how to organise your expectations into tests. The main thing is that the message associated with the test should be informative so that you can quickly narrow down the source of the problem. Try to avoid putting too many expectations in one test - it’s better to have more smaller tests than fewer larger tests

re the discuss TODOs:

#TODO Replace this with http / vcr testing etc
-> will merge into the httptesting branch once we have merged this into dev and update the test there 
#TODO Discuss: should this only be tested within KoboClient?
-> probably again a matter of taste. i moved the test to the test-kobo-client.R file which then feels more like a unit test. testing this in test-kobo.R as well would be more of an integration test i suppose (not that this is a bad thing but our first priority should be unit tests which should happen "as close to the code" as possible from my understanding. 

#TODO Discuss: should we think about custom errors for http stuff? -> probably not
-> agreed. maybe a bit further down the line but i think we'll be busy with the response classes first. :)

**Review by pr130 — APPROVED** [review]

**Comment by pr130** [comment]
> #TODO Replace this with http / vcr testing etc
> -> will merge into the httptesting branch once we have merged this into dev and update the test there

i converted the test into a vcr test and also fixed some things in the testing code.

File: tests/fixtures/kobo-client-get-404.yml
Comment (dimitri-mar) [inline tests/fixtures/kobo-client-get-404.yml:21]: I am not sure how the testing works here, so apologies if what follows do not make sense. 
Should we commit this file? There are dates that I guess they will change at any run.
Should this file be added to gitignore or is it a template for the tests?
This would apply also to  tests/fixtures/kobo-client-403.yml

File: R/kobo.R
Comment (dimitri-mar) [inline R/kobo.R:41]: The optional argument in query is not really helpful: if what we mean is that by default the get function should request output in JSON format, I would include it as an extra argument to the function. In fact, if a user writes a query list now, the optional format will be overridden. 
I would propose 
`function(path, query= None, version = "v2", format="json") `
and then always append to the query a format (in line 42 and 48, or before )

**Review by dimitri-mar — COMMENTED** [review]
Ehi, very nice work! I just included a couple of comments.

File: R/kobo.R
Comment (dimitri-mar) [reply R/kobo.R:41]: Sorry, I could not comment on the line before...

File: tests/fixtures/kobo-client-get-404.yml
Comment (pr130) [reply tests/fixtures/kobo-client-get-404.yml:21]: nope they only have to be re-run when the request you make in the test changes or if you choose so. sorry that i still haven't written the vignette on this, will do this on the weekend. 
so the date only changed because i had to adapt the test a final time and then had to re-run it. i also could've not committed this change but somehow was too lazy to revert the change and just committed it ;)

**Comment by pr130** [comment]
follow up PR: #29

**Pull request merged**
